agent/src/control_interface/filesystem.rs

```diff
@@ -93,7 +95,39 @@ impl FileSystem {
     }
     pub fn make_dir(&self, path: &Path) -> Result<(), FileSystemError> {
         match create_dir_all(path) {
-            Ok(_) => Ok(()),
+            Ok(_) => {
+                if let Some(base_path) = path.parent() {
+                    if !base_path.is_empty() {
+                        let mut current_base_path_permissions = std::fs::metadata(base_path)
+                            .map_err(|err| {
+                                FileSystemError::CreateDirectory(
+                                    path.as_os_str().to_owned(),
+                                    err.kind(),
+                                )
+                            })?
+                            .permissions();
+
+                        // Use 'rwxrwxrwx' permissions on the base folder e.g. /tmp/ankaios
+                        // to allow everyone to read/write from/to it.
+                        // This needs to be re-evaluated again when adding security features!!!
```

Comment:
If the solution already prevents that non-root users can access the pipes then we should delete the hint to re-evaluate this.

Reply:
done